merkledag/utils/utils.go

```diff
@@ -123,23 +153,29 @@ func rmLink(ctx context.Context, ds dag.DAGService, root *dag.Node, path []strin
 		return root, nil
 	}
 
-	nd, err := root.GetLinkedNode(ctx, ds, path[0])
+	nd, err := root.GetLinkedNode(ctx, e.tmp, path[0])
```

Comment:
maybe comment what these two lines do? (i.e. check one first, then the other, and why)